### ISSUE TITLE
ALC has NAN's

### ISSUE THREAD

**Issue by AnthonyLim23** [mantidproject/mantid#35587]
Looking at the last release and comparing the results to this, the problem is coming from the x values having different values. The only code change that has happened in this part of the code base is from the time series changes https://github.com/mantidproject/mantid/commits/main/qt/scientific_interfaces/Muon so I wonder if the changes to log values and time series has caused these values to change. This is something we need to fix.

follow the instructions from https://developer.mantidproject.org/Testing/MuonAnalysis_test_guides/Muon_ALC.html 

when doing the baseline fit an error will occur. If you export the data and inspect the values, near the end of the table is a NAN, the values are different to the previous release too.

**Comment by AnthonyLim23** [comment]
After some more investigating the error is in the calculation of the mean values. If you set up the ALC as described in the testing doc (up to pressing Load). 

Click export results and call the ws "mean"

In the GUI change the "Take log value at" to Max
Press Load, wait for it to complete
Click export results and call the ws "Max"

In the GUI change the "Take log value at" to Min
Press Load, wait for it to complete
Click export results and call the ws "Min"

The below scripts proves that the internal calculation for the mean is incorrect:
```
ma = mtd['Max_Loaded_Data']
mi = mtd['min_Loaded_Data']
me = mtd['mean_Loaded_Data']
mean = (ma.readX(0) + mi.readX(0))/2.
x = me.readX(0)
for k in range(len(ma.readX(0))):
    print(x[k], mean[k])
```

**Comment by AnthonyLim23** [comment]
I have added the high priority label as I think this has come from a change elsewhere in the core and other areas may be impacted too.

**Comment by AnthonyLim23** [comment]
closed by #35609
